django/db/models/sql/compiler.py

```diff
@@ -852,11 +856,12 @@ def execute_sql(self, result_type=MULTI):
             cursor, self.connection.features.empty_fetchmany_value,
             self.col_count
         )
-        if not self.connection.features.can_use_chunked_reads:
+        if not self.connection.features.can_use_chunked_reads and not chunked_fetch:
```

Comment:
Might want to move the `not chunked_fetch` local variable checks before performing a double attribute lookup  with `self.connection.features.can_use_chunked_reads`.